### PR TITLE
[feat] Do not allow chaining of the `-c` option; pass multiple paths as a colon separated list 

### LINKED ISSUE #1120: Make `-c` option accept colon separated paths instead of appending to the check search path
This will address scenarios like the one reported in #1119.

### PR DESCRIPTION
**Pull request by teojgo**
* Deprecate using the checkpath command line option multiple times
  and issue a warning.

* Create unittest for colon separated paths.

* Add the change to the documentation.

Fixes #1120.

### CODE REVIEW

File: reframe/frontend/cli.py [hunk 1]
```diff
@@ -410,7 +410,11 @@ def main():
     # Setup the check loader
     if options.checkpath:
         load_path = []
-        for d in options.checkpath:
+        if len(options.checkpath) > 1:
+            printer.warning("using command line option `--checkpath' "
+                            "multiple times is deprecated, please use "
```
Comment (ekouts) [inline reframe/frontend/cli.py]: Again, I would suggest to warn the user that only the last check path will be taken into account, unless we keep supporting multiple `--checkpath`.

File: docs/running.rst [hunk 2]
```diff
@@ -321,6 +321,17 @@ The ``-c``\ option accepts also regular files. This is very useful when you are
...
+.. note::
+   Using the command line ``-c`` or ``--checkpath`` multiple times is now deprecated and only the last option is going to be taken into account.
+   Multiple paths should be passed as a colon separated string to the above option:
+
+   .. code-block:: bash
+
+      ./bin/reframe -c /path/to/my/first/test.py:/path/to/my/second/test.py -r
+
+
+   .. versionchanged:: 2.22
```
Comment (vkarak) [inline docs/running.rst]:
```suggestion
   .. versionchanged:: 3.0
```

File: docs/running.rst [hunk 3]
```diff
@@ -321,6 +321,17 @@ The ``-c``\ option accepts also regular files. This is very useful when you are
 
    .. versionadded:: 2.12
 
+.. note::
+   Using the command line ``-c`` or ``--checkpath`` multiple times is now deprecated and only the last option is going to be taken into account.
```
[the remark this answers is not in the document]
Comment (vkarak) [reply docs/running.rst]: Also change the "is going to be taken into account." to "will be considered."

File: docs/running.rst [hunk 4]
```diff
@@ -321,6 +321,17 @@ The ``-c``\ option accepts also regular files. This is very useful when you are
...
+   Multiple paths should be passed as a colon separated string to the above option:
```
Comment (vkarak) [inline docs/running.rst]:
```suggestion
   Multiple paths should be passed instead as a colon separated list:
```

File: docs/running.rst [hunk 5]
```diff
@@ -321,6 +321,17 @@ The ``-c``\ option accepts also regular files. This is very useful when you are
...
+
+   .. code-block:: bash
+
+      ./bin/reframe -c /path/to/my/first/test.py:/path/to/my/second/test.py -r
```
Comment (vkarak) [inline docs/running.rst]: Let the second path be a directory, so as to show that you can mix both.

File: reframe/frontend/cli.py [hunk 6]
```diff
@@ -94,7 +94,7 @@ def main():
     # Check discovery options
     locate_options.add_argument(
         '-c', '--checkpath', action='append', metavar='DIR|FILE',
```
Comment (vkarak) [inline reframe/frontend/cli.py]: Make the action `store`.

File: reframe/frontend/cli.py (hunk 1 above)
Comment (vkarak) [reply reframe/frontend/cli.py]: Since we are moving to 3.0, just remove the warning.

File: reframe/frontend/cli.py [hunk 7]
```diff
@@ -94,7 +94,7 @@ def main():
...
-        help='Search for checks in DIR or FILE')
+        help="Search for checks in DIR or FILE, using `:' for multiple paths")
```
Comment (vkarak) [inline reframe/frontend/cli.py]:
```suggestion
        help="Search for checks in DIR or FILE; multiple paths can be separated with `:'")
```

**Comment by vkarak** [comment]
@ekouts Can you also approve this, so that we can merge it?

**Pull request merged**